devicetypes/smartthings/zwave-metering-dimmer.src/zwave-metering-dimmer.groovy

```diff
@@ -160,25 +163,32 @@ def dimmerEvents(physicalgraph.zwave.Command cmd) {
 		result << createEvent(name: "level", value: cmd.value, unit: "%")
 	}
 	if (switchEvent.isStateChange) {
-		result << response(["delay 3000", zwave.meterV2.meterGet(scale: 2).format()])
+		result << response(["delay 3000", meterGet(scale: 2).format()])
 	}
 	return result
 }
 
-def zwaveEvent(physicalgraph.zwave.commands.meterv3.MeterReport cmd) {
-	if (cmd.meterType == 1) {
-		if (cmd.scale == 0) {
-			return createEvent(name: "energy", value: cmd.scaledMeterValue, unit: "kWh")
-		} else if (cmd.scale == 1) {
-			return createEvent(name: "energy", value: cmd.scaledMeterValue, unit: "kVAh")
-		} else if (cmd.scale == 2) {
-			return createEvent(name: "power", value: Math.round(cmd.scaledMeterValue), unit: "W")
-		} else {
-			return createEvent(name: "electric", value: cmd.scaledMeterValue, unit: ["pulses", "V", "A", "R/Z", ""][cmd.scale - 3])
-		}
+def handleMeterReport(cmd)
+{
+    if (cmd.scale == 0) {
+		createEvent(name: "energy", value: cmd.scaledMeterValue, unit: "kWh")
+	} else if (cmd.scale == 1) {
+		createEvent(name: "energy", value: cmd.scaledMeterValue, unit: "kVAh")
+	} else if (cmd.scale == 2) {
+		createEvent(name: "power", value: Math.round(cmd.scaledMeterValue), unit: "W")
 	}
 }
 
+def zwaveEvent(physicalgraph.zwave.commands.meterv3.MeterReport cmd) {
+    log.debug "v3 Meter report: "+cmd
+    handleMeterReport(cmd)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.meterv1.MeterReport cmd) {
```

Comment:
You won't be receiving any meter reports because the list of command classes and versions passed to `parse` specify that we want to parse all meter command class events as v3.